core/src/main/java/org/apache/accumulo/core/fate/FateExecutor.java

```diff
@@ -105,19 +105,26 @@ public FateExecutor(Fate<T> fate, T environment, Set<Fate.FateOperation> fateOps
    */
   protected void resizeFateExecutor(Map<Set<Fate.FateOperation>,Integer> poolConfigs,
       long idleCheckIntervalMillis) {
-    final var pool = transactionExecutor;
     final int configured = poolConfigs.get(fateOps);
-    ThreadPools.resizePool(pool, () -> configured, poolName);
-    synchronized (runningTxRunners) {
-      final int needed = configured - runningTxRunners.size();
-      if (needed > 0) {
-        // If the pool grew, then ensure that there is a TransactionRunner for each thread
+    ThreadPools.resizePool(transactionExecutor, () -> configured, poolName);
+    final int needed = configured - runningTxRunners.size();
+    if (needed > 0) {
+      // If the pool grew, then ensure that there is a TransactionRunner for each thread
+      log.trace("FateExecutor {} needs {} more TransactionRunners", fateOps, needed);
+      synchronized (runningTxRunners) {
```

Comment:
I still think it would be best to keep the entire if-elseif-else locked under runningTxRunners to prevent any potential concurrency issues. This code will lock to get the runningTxRunners.size(), unlock, then lock again under the assumption that runningTxRunners.size() is the same which it might not be